[Chore] Spring Boot 애플리케이션 생성

Cass07 opened the pull request:
## 개요
기존 main branch를 revert하고, 빈 Spring Boot Application 생성
- JPA 모듈을 추가하면서 정상적으로 빈 프로젝트가 실행될 수 있도록 설정값 추가
   - application.yaml 파일을 추가하고 driver-class-name, url, username, password를 추가
   - url, username, password는 환경 변수로 우선 추가

Cass07 committed:
[Chore] Spring Boot 애플리케이션 생성

Spring Boot 빈 애플리케이션 생성
- JPA, DB-Driver 추가
- JPA 초기화를 위해 application.yaml에 DB url, user, password 설정 추가

f-lab-jd commented on build.gradle:
```diff
@@ -0,0 +1,41 @@
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
+    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
+
+    implementation 'org.springdoc:springdoc-openapi-starter-webmvc-ui:2.5.0'
```
이 의존성은 필요한 이유가 무엇인가요~?

Cass07 replied:
- spring-boot-starter-test 의존성은 spring boot 테스트와 관련된 어노테이션을 사용할 수 있게 해 주는 의존성
- junit-platform-launcher 의존성은 JUnit 프레임워크를 사용할 수 있게 해 주는 의존성  


위 두 의존성은 테스트 코드를 작성할 때 쓰이는데, spring-boot-starter-test 패키지가 `testImplementation`으로 정의되어 있어서 `testCompileClassPath`, `testRuntimeClassPath`에 추가되고, (테스트 컴파일 시와, 실행 시 둘다 필요하다는 뜻) 
junit-platform-launcher 의존성은 `testRuntimeOnly`로 정의되어 있기 때문에 `testRuntimeClassPath`에만 추가되야 합니다 (테스트 실행 시에만 필요)  
그런데 찾아보니 spring-boot-starter-test가 하위 라이브러리로 junit-platform-launcher를 가지고 있어서, 34줄의 `testRuntimeOnly 'org.junit.platform:junit-platform-launcher'`는 제거해도 문제가 없을 것으로 생각됩니다. (실제로 IDE의 의존성 관리 탭에서 이미 포함되어있어 제외된다고 출력됩니다)
![image](https://github.com/user-attachments/assets/1121c4da-4e7c-4d1f-a330-cd6c6d309a6c)


- springdoc-openapi-starter-webmvc-ui 의존성은 Springdoc을 Swagger UI로 사용하기 위한 라이브러리인데, 제거하고 추후에 Swagger를 추가할 때 다시 포함하겠습니다.

f-lab-jd commented on src/main/resources/application.yaml:
```diff
@@ -0,0 +1,6 @@
+spring:
+  datasource:
+    driver-class-name: org.mariadb.jdbc.Driver
+    url: ${DB-URL}
```
환경 변수에 대한 주입은 어디에서 이루어지나요~?

Cass07 replied:
우선은 빌드한 jar 파일 실행 시 argument로 주입하고 있습니다
```shell
java -jar knorda-0.0.1-SNAPSHOT.jar -DDB-URL=...
```
추후 CI-CD 과정에서 Github Actions Secret에 환경변수 값을 추가해서 빌드 시 property에 직접 추가할 예정입니다.

f-lab-jd commented on src/test/java/dev/cass/knorda/KnordaApplicationTests.java:
```diff
@@ -0,0 +1,12 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
처음 등장하는 어노테이션이니 저의 컨벤션에 의하면 주석으로 설명이 필요합니다.

Cass07 replied:
누락된 주석 설명을 추가하여 커밋하였습니다

f-lab-jd commented:
정적 분석 검사도 확인해보셨을까요~?

Cass07 committed:
[Chore] Gradle에서 불필요한 패키지 제거

중복 선언되었거나, 현재 사용하지 않는 패키지 종속성 제거
- junit-platform-launcher는 spring-boot-starter-test를 포함하며 같이 포함되기에 제거
- springdoc-openapi-starter-webmvc-ui 는 현재 사용하지 않으므로 제거

Cass07 committed:
[Docs] 신규 어노테이션의 주석 추가

처음 선언한 어노테이션의 설명 주석 추가
- @SpringBootTest, @Test

f-lab-jd reviewed (approved):
👍
